libraries/core-react/src/Popover/Popover.tsx

```diff
@@ -52,30 +60,22 @@ export const Popover = forwardRef<HTMLDivElement, Props>(function Popover(
     return <Container {...props} />
   }
   const anchorRef = useRef<HTMLDivElement>(null)
-  const popoverChildren: PopoverChild | PopoverChild[] = children
-  let anchorElement: PopoverChild
-  const childArray = []
-  if (Array.isArray(popoverChildren)) {
-    for (let i = 0; i < popoverChildren.length; i += 1) {
-      /* 
-      Find anchor element in children to wrap the element together with <PopoverItem/>.
-      Children is required, but user has to wrap the actual anchor with <PopoverAnchor />
-      */
-      const child = popoverChildren[i] as PopoverChild
-      if (child.type && child.type.displayName === 'eds-popover-anchor') {
-        anchorElement = child
-      } else {
-        // Add the remaining children to a new array to display inside <PopoverItem/>
-        childArray.push(child)
+
+  const { anchorElement, childArray } = React.Children.toArray(children).reduce(
+    (acc: PopoverSplit, child): PopoverSplit => {
+      if (isValidElement(child) && child.type === PopoverAnchor) {
+        return {
```

Comment:
using spread here resulted in empty `childArray` so reassignment like this had to be done instead... 🤷‍♂️

Does not work
```jsx
       return {
          anchorElement: child,
         ...acc,
        }
```

Works 🤯
```jsx
       return {
          anchorElement: child,
          childArray: acc.childArray,
        }
```